concepts/0519-goal-codes/README.md

```diff
@@ -0,0 +1,149 @@
+
+### Namespacing
+
+To avoid collision and ambiguity in code values, we need to support namespacing in our goal codes. Since goals are only a coarse-grained alignment mechanism, however, we don't need perfect decentralized precision. Confusion isn't much more than an annoyance; the worst that could happen is that two agents discover one or two steps into a protocol that they're not as aligned as they supposed. They need to be prepared to tolerate that outcome in any case.
+
+Thus, we follow the same general approach that's used in java's packaging system, where organizations and communities use a self-declared prefix for their ecosystem as the leftmost segment or segments of a family of identifiers (goal codes) they manage. Unlike java, though, these need not be tied to DNS in any way. We recommend a single segment namespace that is a unique string, and that is an alias for a URI identifying the origin ecosystem. (In other words, you don't need to start with "com.yourcorp.yourproduct" -- "yourcorp" is probably fine.)
+
+The `aries` namespace alias is reserved for goal codes defined in Aries RFCs. The URI aliased by this name is TBD. See the [Reference section](#reference) for more details.
+
+### Declaring goal codes
+
+#### Standalone RFCs or Similar Sources
+
+Any URI-referencable document can declare famlies or ontologies of goal codes. In the context of Aries, we encourage standalone RFCs for this purpose if the goals seem likely to be relevant in many contexts. Other communities may of course document goal codes in their own specs -- either dedicated to goal codes, or as part of larger topics. The following block is a sample of how we recommend that such goal codes be declared. Note that each code is individually hyperlink-able.
+
+>#### goal codes
+>##### `aries.sell`
+>Sell something. Assumes two parties (buyer/seller)
+>##### `aries.sell.goods.consumer`
+>Sell tangible goods of interest to general consumers.
+>##### `aries.sell.services.consumer`
+>Sell services of interest to general consumers.
+>##### `aries.sell.services.enterprise`
+>Sell services of interest to enterprises.
+
+#### In DIDComm-based Protocol Specs
+
+Occasionally, goal codes may have meaning only within the context of a specific protocol. In such cases, it may be appropriate to declare the goal codes directly in a protocol spec. This can be done using a section of the RFC as described above.
+
+More commonly, however, a protocol will *accomplish* one or more goals (e.g., when the protocol is fulfilling a co-protocol interface), or will require a participant to *identify* a goal at one or more points in a protocol flow. In such cases, the goal codes are probably declared external to the protocol. If they can be enumerated, they should still be referenced (hyperlinked to their respective definitions) in the protocol RFC.
```

Comment:
> a participant to *identify* a goal at one or more points in a protocol flow

In this case the goal codes the participant responds with are merely data - we should not confuse the actual goal of this particular interaction ("myprotocol.collect.goals").

Reply:
> *accomplish* one or more goals

Suggest we make it an array.